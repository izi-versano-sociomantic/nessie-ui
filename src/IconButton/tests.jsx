import React            from 'react';
import { render }       from 'react-testing-library';
import 'jest-dom/extend-expect';

import { IconButton }   from '..';


describe( 'snapshot tests', () =>
{
    test( 'should render IconButton with default props', () =>
    {
        const wrapper = render( <IconButton id = "first" /> );
        // wrapper.debug();
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'should render IconButton with disabled styles', () =>
    {
        const wrapper = render( <IconButton id = "second" isDisabled /> );
        // wrapper.debug();
        expect( wrapper ).toMatchSnapshot();
    } );
} );
