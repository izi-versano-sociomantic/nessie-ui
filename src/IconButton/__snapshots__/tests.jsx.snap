// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests should render IconButton with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="default role__default size__undefined"
        id="first"
        tabindex="0"
        type="button"
      >
        <svg
          class="default role__default size__S"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="default role__default size__undefined"
      id="first"
      tabindex="0"
      type="button"
    >
      <svg
        class="default role__default size__S"
      />
    </button>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`snapshot tests should render IconButton with disabled styles 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="default disabled role__default size__undefined"
        disabled=""
        id="second"
        tabindex="0"
        type="button"
      >
        <svg
          class="default role__default size__S"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="default disabled role__default size__undefined"
      disabled=""
      id="second"
      tabindex="0"
      type="button"
    >
      <svg
        class="default role__default size__S"
      />
    </button>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
